### ISSUE TITLE
DNS_Zones naming format

### ISSUE THREAD

**Issue by jlagermann** [F5Networks/f5-appsvcs-extension#118]
##### ISSUE TYPE
 - Bug Report

##### AS3 BUILD/ VERSION
```
3.13.0
```

##### BIGIP VERSION
```
Sys::Version
Main Package
  Product     BIG-IP
  Version     15.0.0
  Build       0.0.39
  Edition     Final
  Date        Fri May 10 03:27:47 PDT 2019
```

##### SUMMARY

```
DNS Zone names are required to be the root domain of the zone (i.e. example.com).  AS3 will not allow "." to be part of the name.  It give the following error:

Configures a DNS zone
String does not match the pattern of "^[A-Za-z][0-9A-Za-z_]*$".
```

##### STEPS TO REPRODUCE
```
{
    "$schema": "https://raw.githubusercontent.com/F5Networks/f5-appsvcs-extension/master/schema/latest/as3-schema.json",
    
    "class": "AS3",
    "action": "deploy",
    "persist": true,
    "declaration": {    
        "class": "ADC",
        "updateMode": "selective",
        "schemaVersion": "3.13.0",
        "id": "DNS",
        "Sample_non_http_03": {
            "class": "Tenant",
            "TEST_DNS_Zone": {
                "class": "Application",
                "template": "generic",
                "example.com": {
                    "class": "DNS_Zone",
                    "label": "dnsZone",
                    "remark": "DNS Zone for example.com",
                    "dnsExpress": {
                        "enabled": true,
                        "nameserver": {
                            "use": "dnsNameserverZone"
                        },
                        "notifyAction": "consume",
                        "allowNotifyFrom": [
                            "10.1.1.1"
                        ],
                        "verifyNotifyTsig": false
                    },
                    "responsePolicyEnabled": true,
                    "serverTsigKey": {
                        "use": "tsigKeyZone"
                    },
                    "transferClients": [
                        {
                            "use": "dnsNameserverZone"
                        }
                    ]
                },
                "dnsNameserverZone": {
                    "class": "DNS_Nameserver",
                    "label": "dnsNameserverZone",
                    "address": "10.1.1.2",
                    "port": 53,
                    "routeDomain": {
                        "bigip": "/Common/0"
                    },
                    "tsigKey": {
                        "use": "tsigKeyZone"
                    }
                },
                "tsigKeyZone": {
                    "class": "DNS_TSIG_Key",
                    "label": "tsigKeyZone",
                    "remark": "TSIG Key test",
                    "algorithm": "hmacmd5",
                    "secret": {
                        "ciphertext": "ZjVmNQ==",
                        "miniJWE": true,
                        "protected": "eyJhbGciOiJkaXIiLCJlbmMiOiJub25lIn0",
                        "ignoreChanges": true,
                        "allowReuse": false
                    }
                }
            }
        },
        "DNS_PROFILE_1": {
            "class": "Tenant", 
            "DNS_PROFILE_1": {
                "class": "Application",
                "template": "udp",
                "serviceMain": {
                    "class": "Service_UDP",
                    "virtualPort": 80,
                    "virtualAddresses": [
                        "198.19.192.210"
                    ],
                    "profileUDP": {
                        "use": "profileUdp"
                    },

                    "profileDNS": {
                        "use": "profileDnsHW"
                    }
                },
                "profileDnsHW": {
                    "class": "DNS_Profile",
                    "label": "profileDnsHW",
                    "remark": "DNS Profile test",
                    "parentProfile": {
                        "bigip": "/Common/dns"
                    },
                    "rapidResponseEnabled": false,
                    "rapidResponseLastAction": "allow",
                    "hardwareQueryValidationEnabled": true,
                    "hardwareResponseCacheEnabled": true,
                    "dnssecEnabled": false,
                    "globalServerLoadBalancingEnabled": false,
                    "dnsExpressEnabled": false,
                    "cacheEnabled": false,
                    "dns64Mode": "secondary",
                    "dns64Prefix": "0:0:0:0:0:0:0:0",
                    "dns64AdditionalSectionRewrite": "any",
                    "unhandledQueryAction": "drop",
                    "localBindServerEnabled": false,
                    "zoneTransferEnabled": true,
                    "recursionDesiredEnabled": false,
                    "securityEnabled": false,
                    "loggingEnabled": false,
                    "statisticsSampleRate": 20
                },
                "profileUdp": {
                    "class": "UDP_Profile",
                    "datagramLoadBalancing": true
                }

            }
        }
    }
}


 
```

##### EXPECTED RESULTS
```
create a DNS zone named example.com
```

##### ACTUAL RESULTS
```
error, zone name string does not match the pattern of "^[A-Za-z][0-9A-Za-z_]*$".
```

**Comment by focrensh** [comment]
AT-629 used for internal tracking.

**Comment by DumpySquare** [comment]
I submit that the naming filters all be loosened to this:  ^[0-9A-Za-z_.-]*$

**Comment by dstokesf5** [comment]
This was resolved in release 3.16.0.
